gui/src/main/scala/co/uproot/abandon/RegReport.scala

```diff
@@ -30,34 +36,49 @@ object RegUIReport extends UIReport {
     new TreeView(reportRoot) {
       styleClass += styleClassName
       onKeyTyped = { e: KeyEvent =>
-        if (e.character equals "\r") {
-          val selectedItemOpt = selectionModel().getSelectedItems().headOption
-          selectedItemOpt foreach { selectedItem =>
-            // println(e)
-            val txStage = new Stage() {
-              scene = new Scene(800, 600) {
-                root = new ScrollPane {
-                  content = TxnUIReport.mkTxnView(getNestedTxns(selectedItem))
-                }
-                stylesheets += "default_theme.css"
-              }
-              initModality(Modality.ApplicationModal)
-              title = "Transactions"
+        if (e.character.equals("\r") && !selectionModel().isEmpty) {
+          // println(e)
+          selectedItem = selectionModel().getSelectedItem
+          // show all transactions of the parent of the selected item if the latter has any transactions
+          // this will allow for an update of the transaction view when an input file changes
+          if (selectedItem.getValue.txns.nonEmpty) selectedItem = selectedItem.getParent
```

Comment:
Cool.. you can use the `TransactionFilter` trait defined [here](https://github.com/hrj/abandon/blob/e938c43e1422764d3b48aab9cab8eb2192646295/base/src/main/scala/co/uproot/abandon/Ast.scala#L127).